src/vanilla/atom.ts

```diff
@@ -5,9 +5,7 @@ type Setter = <Value, Args extends unknown[], Result>(
   ...args: Args
 ) => Result
 
-type SetAtom<Args extends unknown[], Result> = <A extends Args>(
-  ...args: A
-) => Result
```

Comment:
After checking the usage of `SetAtom`, it seems that the `<A extends Args>` part is unnecessary.